src/main/java/com/ericsson/ei/subscription/SubscriptionValidator.java

```diff
@@ -70,29 +75,36 @@ public static void validateSubscription(Subscription subscription) throws Subscr
     }
 
     /**
-     * Validation of subscriptionName parameter Throws SubscriptionValidationException if validation of
-     * the parameter fails due to wrong format of parameter.
+     * Validation of subscriptionName parameter Throws
+     * SubscriptionValidationException if validation of the parameter fails due to
+     * wrong format of parameter.
      *
      * @param subscriptionName
      */
     private static void validateSubscriptionName(String subscriptionName) throws SubscriptionValidationException {
-        // When this regExp need to be changed then remember to change the one in the
-        // back-end (invalidSubscriptionNameRegex in subscription.js), which do the same
-        // invalid subscription name check. The two
-        // regEx always need to be the same for ensuring the same check.
-        // /(\W)/ Is a regEx that matches anything that is not [A-Z,a-z,0-8] and _.
-        String invalidSubscriptionNameRegex = "(\\W)";
+        // Here the regEx used is :/(\W)/ that matches anything that is not
+        // [A-Z,a-z,0-8] and _.
+        String invalidSubscriptionNameRegex = null;
+        try {
+            invalidSubscriptionNameRegex = RegExProvider.getRegExs().getString("invalidName");
+        } catch (JSONException | IOException e) {
+            LOGGER.error("Error message: " + e.getMessage(), e);
+        }
+
         if (subscriptionName == null) {
             throw new SubscriptionValidationException("Required field SubscriptionName has not been set");
+        } else if (invalidSubscriptionNameRegex == null || invalidSubscriptionNameRegex.isEmpty()) {
+            throw new SubscriptionValidationException(
```

Comment:
Move this regex check to it´s own function that may be called for both name validation and email validation, thus there is not need to have duplicated code.